bridge/elrond/client.go

```diff
@@ -73,35 +75,39 @@ func NewClient(config bridge.Config) (*Client, string, error) {
 		return nil, "", err
 	}
 
-	return &Client{
+	client := &Client{
 		proxy:         proxy,
 		bridgeAddress: config.BridgeAddress,
 		privateKey:    privateKey,
 		address:       address.AddressAsBech32String(),
-		nonce: func() (uint64, error) {
+		log:           log,
+	}
+
+	go func() {
```

Comment:
Maybe we should make this aware of the context, we do pass the context as it's needed by the eth sdk.

Reply:
agree, I would add a context with cancel function + an exported Close method as to be able to control this go routine.